xopt/generators/bayesian/models/time_dependent.py

```diff
@@ -31,6 +38,30 @@ def build_model(
         new_input_bounds = deepcopy(input_bounds)
         new_input_bounds["time"] = [min_t, max_t]
 
+        # set covar modules if not specified -- use SpectralMixtureKernel for time axis
+        # see Kuklev, N., et al. "Online accelerator tuning with adaptive bayesian optimization."
+        # Proc. NAPAC 22 (2022): 842.
+        if len(self.covar_modules) == 0 and self.use_spectral_mixture_kernel:
+            covar_modules = {}
+            for name in outcome_names:
+                if len(input_names) == 1:
+                    matern_dims = [0]
+                else:
+                    matern_dims = tuple(range(len(input_names)))
+                time_dim = [len(input_names)]
+
+                matern_kernel = MaternKernel(
+                    nu=2.5,
+                    active_dims=matern_dims,
+                    lengthscale_prior=GammaPrior(3.0, 6.0),
+                )
+
+                covar_modules[name] = ProductKernel(
+                    SpectralMixtureKernel(num_mixtures=5, active_dims=time_dim),
```

Comment:
GPyTorch recommends to initialize SMK from data. For example, from code in the paper above:
```
    if enable_time:
        smk = SpectralMixtureKernel(num_mixtures=num_smk_mixtures, ard_num_dims=1, active_dims=(dx.shape[-1] - 1,))
        smk.initialize_from_data(dx, dy)

        covar_module = ProductKernel(smk, regular_kernel)
        subset_batch_dict = {
            "likelihood.noise_covar.raw_noise": -2,
            "covar_module.base_kernel.kernels.1.raw_outputscale": -1,
            "covar_module.base_kernel.kernels.1.base_kernel.raw_lengthscale": -3,
        }
```

The subset dict part is not used by us, but initialization should be helpful. Looking at source of that heuristic, it has some statistical tests and shouldn't be a performance problem. So, maybe just do it here as well?

Reply:
Also, I found 3 components a more stable choice generally.